apps/browser/qml/pages/components/AddHomePageDialog.qml

```diff
@@ -0,0 +1,63 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.0
+import Sailfish.Silica 1.0
+import Sailfish.Policy 1.0
+import org.nemomobile.configuration 1.0
+
+Dialog {
+    id: root
+    property alias text: textField.text
+    property alias placeholderText: textField.placeholderText
+    property alias description: textField.description
+
+    function containsSpaces(text) {
+       return /^\s+|\s+$|\s*(\s)/.test(text) && text.length > 0
+    }
+
+    canAccept: (textField.text !== "" && !textField.errorHighlight)
+
+    DialogHeader {
+        id: header
+        //: Accept button text for adding a home page adress
+        //% "OK"
+        acceptText: qsTrId("sailfish_browser-he-ok")
+    }
+
+    onAccepted: {
+        homePageConfig.value = textField.text || "about:blank"
+    }
+
+    TextField {
+        id: textField
+
+        width: parent.width
+        anchors.top: header.bottom
+        enabled: AccessPolicy.browserEnabled
+        text: homePageConfig.value === "about:blank" ? "" : homePageConfig.value
+        //% "Enter website address that should be shown when browser is opened with no tabs to load."
+        description: qsTrId("sailfish_browser-he-home_page_website-address")
+        //% "Website address"
+        placeholderText: qsTrId("settings_browser-la-website_address")
+        label: placeholderText
+        inputMethodHints: Qt.ImhNoPredictiveText | Qt.ImhNoAutoUppercase | Qt.ImhUrlCharactersOnly
+        EnterKey.iconSource: "image://theme/icon-m-enter-close"
+        EnterKey.onClicked: focus = false
+        errorHighlight: containsSpaces(text)
```

Comment:
It's customary to switch out the description with some info about why there's an error in case the dialog can't be accepted. For example, something like "Spaces can't be included in the URL".

Is it necessary to exclude spaces from the URL?

Reply:
@llewelld  Done.  Added by "Spaces can't be included in the URL", but there was no corresponding inscription in the design. 
`Is it necessary to exclude spaces from the URL?` Сombobox will have an extra space at the beginning of the line, and desktop browsers remove spaces in the address automatically.

Reply:
Could you use `trim()` to remove the extra spaces instead for the combo box instead (or maybe I'm misunderstanding)?

It's just that we allow spaces in the address bar, and in favourites, so it would seem reasonable to allow them here too (otherwise the user has to manually change each of them to a %20 instead).

Reply:
@llewelld  regexp finds all spaces, if, for example, the user enters https: // goo gle.com, then when the browser starts, not the page `https: // google.com` will open, but a search query, in the field of which there will be` https: // goo gle .com`. It checks and warns the user that this should not be done.